src/org/labkey/test/tests/DataClassFolderExportImportTest.java

```diff
@@ -0,0 +1,351 @@
+
+        DataClassDefinition testType = new DataClassDefinition(testDataClass).setFields(DataClassAPIHelper.dataClassTestFields());
+
+        TestDataGenerator testDgen = DataClassAPIHelper.createEmptyDataClass(subfolderPath, testType);
+
+        testDgen.addCustomRow(Map.of("Name", "class1", "intColumn", 1, "decimalColumn", 1.1, "stringColumn", "one"));
+        testDgen.addCustomRow(Map.of("Name", "class2", "intColumn", 2, "decimalColumn", 2.2, "stringColumn", "two"));
+        testDgen.addCustomRow(Map.of("Name", "class3", "intColumn", 3, "decimalColumn", 3.3, "stringColumn", "three"));
+        testDgen.addCustomRow(Map.of("Name", "class4", "intColumn", 4, "decimalColumn", 4.4, "stringColumn", "four"));
+        testDgen.addCustomRow(Map.of("Name", "class5", "intColumn", 5, "decimalColumn", 5.5, "stringColumn", "five"));
+        testDgen.insertRows();
+
+        Integer expectedRecordCount = testDgen.getRowCount();
+
+        PortalHelper portalHelper = new PortalHelper(this);
+        portalHelper.addWebPart("Data Classes");
+        portalHelper.addWebPart("Experiment Runs");
+
+        DataRegionTable sourceRunsTable = DataRegionTable.DataRegion(getDriver()).withName("Runs").waitFor();
+        List<String> runNames = sourceRunsTable.getColumnDataAsText("Name");
+
+        clickAndWait(Locator.linkWithText(testDataClass));
+        DataRegionTable sourceTable = DataRegionTable.DataRegion(getDriver()).withName("query").waitFor();
+        List<Map<String, String>> sourceRowData = new ArrayList<>();
+        for (int i=0; i<_attachments.size(); i++)
+        {
+            sourceTable.clickEditRow(i);
+            setFormElement(Locator.input("quf_attachmentColumn"), _attachments.get(i));
+            clickButton("Submit");
+        }
+        for (int i=0; i<expectedRecordCount; i++)
+        {
+            sourceRowData.add(sourceTable.getRowDataAsMap(i));
+        }
+
+        // act - export to file and import the file to our designated import directory
+        goToFolderManagement()
+                .goToExportTab();
+        File exportedFolderFile = doAndWaitForDownload(()->findButton("Export").click());
+
+        goToProjectFolder(IMPORT_PROJECT_NAME, importFolder);
+        importFolderFromZip(exportedFolderFile, false, 1);
+        goToProjectFolder(IMPORT_PROJECT_NAME, importFolder);
+
+        List<String> importedRunNames = DataRegionTable.DataRegion(getDriver()).withName("Runs").waitFor()
+                .getColumnDataAsText("Name");
+        for (String sourceRun : runNames)
+        {
+            assertThat("expect all runs to come through", importedRunNames, hasItems(sourceRun));
+        }
+
+        // we expect the sample types and experiment runs webparts to come across along with the folder import
+        clickAndWait(Locator.linkWithText(testDataClass));
+        DataRegionTable destSamplesTable = DataRegionTable.DataRegion(getDriver()).withName("query").waitFor();;
+
```

Comment:
seems fine but the webpart roundtripping wasn't specific to the data class export/import